### ISSUE TITLE
Insecure temporary file usage

### ISSUE THREAD

**Issue by jrusnack** [rear/rear#607]
Simply grepping for /tmp in source code gives a lot of hits, some of which are dangerous, e.g.:

usr/share/rear/verify/DP/default/50_select_dp_restore.sh:
```
test -f /tmp/dp_list_of_sessions.in && rm -f /tmp/dp_list_of_sessions.in
```
Would be nice if all occurences of hardcoded /tmp/.. were replaced with using `mktemp`.

**Comment by gdha** [comment]
See RH Bugzilla https://bugzilla.redhat.com/show_bug.cgi?id=1239009

**Comment by gdha** [comment]
There are 2 more reports:
- https://bugzilla.redhat.com/show_bug.cgi?id=1238843 (for RHEL 7.2)
- https://bugzilla.redhat.com/show_bug.cgi?id=1245189 (for f22)

**Comment by gdha** [comment]
remark: RH Bugzilla 1245189 has been closed as it is duplicate of 1239009.

**Comment by phracek** [comment]
I have a question related to this issue. 
I have found that $TMP_DIR is setup by usr/bin/rear binary via mktemp -d -t command line.
I have found that there are other binaries/scripts which have still /tmp definitions.
Like:
share/rear/output/TSM/default/95_dsmc_save_result_files.sh [12]
share/rear/output/NSR/default/95_nsr_save_result_files.sh [12]
share/rear/lib/validate-workflow.sh [118,121]
share/rear/finalize/GNU/Linux/16_rename_diskbyid.sh [23]
share/rear/rescue/default/91_copy_logfile.sh [3] ???

**Comment by phracek** [comment]
Shall I send you PR for some of these scripts?

**Comment by gdha** [comment]
@phracek An extra PR :+1: fine for me.

**Comment by gdha** [comment]
@phracek I believe we have fixed all `/tmp/` related stuff in rear 1.17.2 - I will make a release beginning of next week
